web-admin/src/features/public-urls/form-utils.ts

```diff
@@ -43,3 +45,84 @@ export function convertDateToMinutes(date: string) {
   const diff = future.getTime() - now.getTime();
   return Math.floor(diff / 60000);
 }
+
+/**
+ * Returns the serialized *sanitized* `state` for the current dashboard.
+ * It removes all state that refers to fields that will be hidden, like filters, pivot chips, and visible field keys.
+ * This ensures we do not leak hidden information to the URL recipient.
+ */
+export function getSanitizedDashboardStateParam(
+  dashboard: MetricsExplorerEntity,
+  metricsViewFields: string[] | undefined,
+): string {
+  // If no metrics view fields are specified, everything is visible, and there's no need to sanitize
+  if (!metricsViewFields) return getProtoFromDashboardState(dashboard);
+
+  // Else, explicitly add the sanitized state that we want to remember.
+  const sanitizedDashboardState = {
+    // Remove any measures not specified in the metrics view fields
+    visibleMeasureKeys: new Set(
+      [...dashboard.visibleMeasureKeys].filter((measure) =>
+        metricsViewFields?.includes(measure),
+      ),
+    ),
+    allMeasuresVisible: dashboard.allMeasuresVisible,
+    // Remove any dimensions not specified in the metrics view fields
+    visibleDimensionKeys: new Set(
+      [...dashboard.visibleDimensionKeys].filter((dimension) =>
+        metricsViewFields?.includes(dimension),
+      ),
+    ),
+    allDimensionsVisible: dashboard.allDimensionsVisible,
+    leaderboardMeasureName: dashboard.leaderboardMeasureName,
+    dashboardSortType: dashboard.dashboardSortType,
+    sortDirection: dashboard.sortDirection,
+    // Remove the where filter
+    // whereFilter: dashboard.whereFilter,
+    havingFilter: dashboard.havingFilter,
+    dimensionThresholdFilters: dashboard.dimensionThresholdFilters,
+    dimensionFilterExcludeMode: dashboard.dimensionFilterExcludeMode,
+    // There's no need to share filters-in-progress
+    // temporaryFilterName: dashboard.temporaryFilterName,
+    selectedTimeRange: dashboard.selectedTimeRange,
+    selectedScrubRange: dashboard.selectedScrubRange,
+    // There's no need to share the user's previous scrub range
+    // lastDefinedScrubRange: dashboard.lastDefinedScrubRange,
+    selectedComparisonTimeRange: dashboard.selectedComparisonTimeRange,
+    // When TDD, we remove the selected comparison dimension (because, if filtered, it's locked & hidden)
+    selectedComparisonDimension:
+      dashboard.activePage === DashboardState_ActivePage.TIME_DIMENSIONAL_DETAIL
+        ? undefined
+        : dashboard.selectedComparisonDimension,
+    // We do not support sharing the dimension table page (because, if filtered, the dimension is locked & hidden)
+    activePage:
+      dashboard.activePage === DashboardState_ActivePage.DIMENSION_TABLE
```

Comment:
Would this not block users from creating a url with dimension table? Especially if the active dimension is not filtered on? Same applies to TDD.